lib/workers/branch/index.ts

```diff
@@ -342,97 +342,115 @@ export async function processBranch(
-              {
-                cmd,
-                allowedPostUpgradeCommands: config.allowedPostUpgradeCommands,
-              },
-              'Post-upgrade task did not match any on allowed list'
-            );
-          } else {
-            const compiledCmd = config.allowPostUpgradeCommandTemplating
-              ? template.compile(cmd, config)
-              : cmd;
+          for (const cmd of commands) {
+            if (
+              !config.allowedPostUpgradeCommands.some((pattern) =>
+                regEx(pattern).test(cmd)
+              )
+            ) {
+              logger.warn(
+                {
+                  cmd,
+                  allowedPostUpgradeCommands: config.allowedPostUpgradeCommands,
+                },
+                'Post-upgrade task did not match any on allowed list'
+              );
+            } else {
+              const compiledCmd = config.allowPostUpgradeCommandTemplating
+                ? template.compile(cmd, upgrade)
+                : cmd;
 
-            logger.debug({ cmd: compiledCmd }, 'Executing post-upgrade task');
+              logger.debug({ cmd: compiledCmd }, 'Executing post-upgrade task');
 
-            const execResult = await exec(compiledCmd, {
-              cwd: config.localDir,
-            });
+              const execResult = await exec(compiledCmd, {
+                cwd: config.localDir,
+              });
 
-            logger.debug(
-              { cmd: compiledCmd, ...execResult },
-              'Executed post-upgrade task'
-            );
+              logger.debug(
+                { cmd: compiledCmd, ...execResult },
+                'Executed post-upgrade task'
+              );
+            }
           }
-        }
 
-        const status = await getRepoStatus();
+          const status = await getRepoStatus();
 
-        for (const relativePath of status.modified.concat(status.not_added)) {
-          for (const pattern of fileFilters) {
-            if (minimatch(relativePath, pattern)) {
-              logger.debug(
-                { file: relativePath, pattern },
-                'Post-upgrade file saved'
-              );
-              const existingContent = await readLocalFile(relativePath);
-              config.updatedArtifacts.push({
-                name: relativePath,
-                contents: existingContent,
-              });
+          for (const relativePath of status.modified.concat(status.not_added)) {
+            for (const pattern of fileFilters) {
+              if (minimatch(relativePath, pattern)) {
+                logger.debug(
+                  { file: relativePath, pattern },
+                  'Post-upgrade file saved'
+                );
+                const existingContent = await readLocalFile(relativePath);
+                const existingUpdatedArtifacts = config.updatedArtifacts.find(
+                  (ua) => ua.name === relativePath
+                );
+                if (!existingUpdatedArtifacts) {
```

Comment:
```suggestion
                if (existingUpdatedArtifacts) {
```

Please flip around this if/else so that there's no negation here